src/Inputs/Slider.tsx

```diff
@@ -1,79 +1,456 @@
-import React from 'react';
-import styled from 'styled-components';
-import { LabelLayout, LabelLayoutProps } from '../Fragments';
+import React, {
+    useState,
+    useLayoutEffect,
+    useRef,
+    RefObject,
+    MouseEvent,
+    useMemo,
+} from 'react';
+import styled, { DefaultTheme } from 'styled-components';
+import { LabelLayout, LabelLayoutProps } from '@Layouts';
+import { Popup } from '../Containers/Popup';
+
+export interface MarkProps {
+    key: number;
+    mark: string;
+}
+
+export interface ResultValues {
+    lowValue: number;
+    highValue: number;
+}
 
 export interface SliderProps extends LabelLayoutProps {
+    result?: ResultValues;
     max?: number;
     min?: number;
     step?: number;
     disabled?: boolean;
-    value?: number;
-    trackColor?: string;
-    buttonColor?: string;
+    hasTwoInputs?: boolean;
+    marks?: MarkProps[];
+    onChange?: Function;
+    valueFinish?: number;
+    valueStart?: number;
+    hasRail?: boolean;
+    left?: number;
+    right?: number;
+    hasPopup?: boolean;
+    popupLeft?: number;
+    popupTop?: number;
+    popupWidth?: number;
+    popupHeight?: number;
+    theme?: DefaultTheme;
 }
 
 export const Slider: React.FunctionComponent<SliderProps> = ({
+    disabled = false,
+    marks,
+    min = 0,
+    max = 100,
+    hasRail,
+    hasTwoInputs = false,
+    valueFinish = max,
```

Comment:
@sahandfarahani What does this variable mean?

Reply:
I think that's supposed to be one of the initial values (valueStart being the other)

Reply:
yes these are Initial Position Values of the knobs